casper/src/main/resources/MultiSigRevVault.rho

```diff
@@ -0,0 +1,263 @@
+              retU!((true, map.get(box)))
+            } else {
+              retU!((false, "Invalid box"))
+            }
+          }
+        } |
+        ret!((*sealer, *unsealer))
+      }
+    } |
+
+    contract MultiSigRevVault(@"deployerAuthKey", deployerId, ret) = {
+      new DeployerIdOps(`rho:rchain:deployerId:ops`),
+          revAddrCh,
+          deployerPubKeyBytesCh in {
+        DeployerIdOps!("pubKeyBytes", *deployerId, *deployerPubKeyBytesCh) |
+        for (@deployerPubKeyBytes <- deployerPubKeyBytesCh) {
+          _fromPublicKey!(deployerPubKeyBytes, *ret)
+        }
+      }
+    } |
+
+    contract _fromPublicKey(@publicKey, ret) = {
+      new RevAddress(`rho:rev:address`), revAddrCh in {
+        RevAddress!("fromPublicKey", publicKey, *revAddrCh)|
+        for (@revAddress <- revAddrCh) {
+          @AuthKey!("make", (*_multiSigRevVault, revAddress), *ret)
+        }
+      }
+    } |
+
+    // MultiSig
+    // Returns (false, errorMsg) or (true, (multiSigVault, revAddress, revVault))
+    contract MultiSigRevVault(@"create", @publicKeys, @unsealers, @quorumSize, ret) = {
+      if (quorumSize > publicKeys.length() + unsealers.length() or
+          quorumSize <= 0) {
+        ret!((false, "invalid quorum size"))
+      } else {
+        new unf, authKeyCh, authPubKeysCh, RevAddress(`rho:rev:address`), revAddrCh in {
+          RevAddress!("fromUnforgeable", *unf, *revAddrCh) |
+          @RevVault!("unforgeableAuthKey", *unf, *authKeyCh) |
+          @ListOps!("map", publicKeys, *_fromPublicKey, *authPubKeysCh) |
+          for (@revVaultAuthKey <- authKeyCh; @authPubKeys <- authPubKeysCh; @revAddr <- revAddrCh) {
+            new multiSig, nonceStore, nonceMapStore, revVaultCh, verify in {
+              @RevVault!("findOrCreate", revAddr, *revVaultCh) |
+              for (@maybeRevVault <- revVaultCh) {
+                match maybeRevVault {
+                  (false, _) => { ret!(maybeRevVault) }
+                  (true, revVault) => {
+                    ret!((true, (*multiSig, revAddr, revVault))) |
+                    // Nonce starts at zero
+                    @(*multiSig, *nonceStore)!(0) |
+
+                    // Map from (revAddress, nonce) to
+                    //   (true, confirmationSet)
+                    //   (false, "transaction complete")
+                    @(*multiSig, *nonceMapStore)!({}) |
+
+                    contract multiSig(@"balance", ret) = {
+                      @revVault!("balance", *ret)
+                    } |
+
+                    contract verify(@auth, idxAuthPubKeysCh, unsealedCh) = {
+                      new unsealerTest in {
+                        // Check whether auth is in the list of authPubKeys
+                        @ListOps!("indexOf", authPubKeys, auth, *idxAuthPubKeysCh) |
+                        // Check whether auth is a sealed tuple
+                        @ListOps!("partialFold", unsealers, (-1, Nil), *unsealerTest, *unsealedCh) |
```

Comment:
Why not make it `(false, Nil)`?

Reply:
The first element in the pair is the index of the unsealer in the list.  The value -1 is standard for "not in the list". https://www.scala-lang.org/api/2.12.3/scala/collection/immutable/StringOps.html#indexOf(elem:Char,from:Int):Int